sharpener/sharp_modules/spec_ex.py

```diff
@@ -76,47 +79,50 @@ def abs_ex(cfg_par):
-					ra_deg_angle  = Angle(np.rad2deg(source.pos.ra) * u.deg)
-					dec_deg_angle = Angle(np.rad2deg(source.pos.dec) * u.deg)
-					ra_hms = ra_deg_angle.to_string(unit=u.hourangle, sep=':')
-					dec_dms = dec_deg_angle.to_string(unit=u.degree, sep=':')
-					J2000_name.append('J{:s}{:s}{:s}'.format(ra_hms.replace(':', ''),
-															'+' if source.pos.dec > 0.0 else '-',
-									  						dec_dms.replace(':', '')))
-					ra.append(ra_hms)
-					dec.append(dec_dms)
-					flux_cont.append(source.flux.I)
-			src_id = np.arange(0,len(ra)+1,1)
-
-        elif cfg_par['source_finder'].get('enable',False) == True:
-
-            src_list_csv = cfg_par['general']['absdir']+'mir_src_sharpener.csv'
+            catalogName = cfg_par[key].get('catalog', 'NVSS')
+            if catalogName == 'NVSS':
+                catalog_table = str(cfg_par['general'].get('absdir')) + 'cat_src_sharpener.txt'
+                tab = ascii.read(catalog_table)
+                J2000_name = tab['NVSS']
+                ra = tab['RAJ2000']
+                dec = tab['DEJ2000']
+                flux_cont = tab['S1.4']
+
+            elif catalogName == 'PYBDSF':
+                J2000_name, ra, dec, flux_cont = [], [], [], []
+                import Tigger
+                from astropy import units as u
+                from astropy.coordinates import Angle
+                catalog_table = '{:s}{:s}'.format(cfg_par['general'].get('workdir'),
+                                                  cfg_par['source_catalog'].get('catalog_file'))
+                model = Tigger.load(catalog_table)
+                sources = model.sources
+                for source in sources:
+                    ra_deg_angle  = Angle(np.rad2deg(source.pos.ra) * u.deg)
+                    dec_deg_angle = Angle(np.rad2deg(source.pos.dec) * u.deg)
+                    ra_hms = ra_deg_angle.to_string(unit=u.hourangle, sep=':')
+                    dec_dms = dec_deg_angle.to_string(unit=u.degree, sep=':')
+                    J2000_name.append('J{:s}{:s}{:s}'.format(ra_hms.replace(':', ''),
+                                                            '+' if source.pos.dec > 0.0 else '-',
+                                                            dec_dms.replace(':', '')))
+                    ra.append(ra_hms)
+                    dec.append(dec_dms)
+                    flux_cont.append(source.flux.I)
+            src_id = np.arange(0,len(ra)+1,1)
+
+        elif os.path.exists(src_list_csv):
+
             # open file
             src_list_vec = ascii.read(src_list_csv)
             J2000_name = np.array(src_list_vec['J2000'],dtype=str)
             ra = np.array(src_list_vec['ra'],dtype=str)
             dec = np.array(src_list_vec['dec'],dtype=str)
             flux_cont = np.array(src_list_vec['peak'],dtype=float)
             src_id = src_list_vec['ID']
-
+        
+        else:
+            print '!!!! catalog of sources does not exist. Enable source_catalog or source_finder first'
```

Comment:
This print statement should just have parenthesis since we will migrate to py3 soonest.